pkg/executables/cmk.go

```diff
@@ -42,13 +42,13 @@ func (c *Cmk) Close(ctx context.Context) error {
 	return nil
 }
 
-func (c *Cmk) ValidateTemplatePresent(ctx context.Context, domainId string, zoneId string, account string, template v1alpha1.CloudStackResourceRef) error {
+func (c *Cmk) ValidateTemplatePresent(ctx context.Context, domainId string, zoneId string, account string, template v1alpha1.CloudStackResourceIdentifier) error {
```

Comment:
Is there an open ticket to refactor `cmk.go`? Executables shouldn't be performing validations, they should just provide programmatic access to atomic APIs for the technology they represent.

Reply:
Thanks for the comment. My reasoning for this verbiage was that the binary `cmk` doesn't expose a "get" operation - only list, with filters, which is what we are using. I couldn't call it "GetTemplate" since that's not really what cmk is doing. The only "validation" taking place here is ensuring there is exactly 1 result from the list operation, which doesn't feel like application level business logic to me. I'm inclined to keep the nomenclature/implementation as is